backend/src/main/java/de/neuefische/app/spotify/SpotifyApiController.java

```diff
@@ -40,13 +44,28 @@ public void callbackUrl(@RequestParam String code)  {
                 request,
                 SpotifyResponse.class
         );
+
+        ResponseEntity<SpotifyUser> userResponse = restTemplate.exchange(
+                "https://api.spotify.com/v1/me",
+                HttpMethod.GET,
+                new HttpEntity<>(createHeaders(accessTokenResponse.getBody().accessToken())),
+                SpotifyUser.class
+        );
+
+        userService.saveUser(new UserDocument(userResponse.getBody().email(), userResponse.getBody().id(), userResponse.getBody().name(), null, null));
```

Comment:
I'm not sure if this should be part of this story, but you need to distinguish between users that need to be created and users that were already saved in the database and are only logging in.

Reply:
I was going to implement that later on in UserService-saveUser as an inquery that checks if user already exist in the db